ToolBox/FxcmDownloader/FxcmDataDownloader.cs

```diff
@@ -0,0 +1,422 @@
+            // from 0-11 (i.e. Jan == 0)
+            return new DateTime(cal.get(Calendar.YEAR),
+                                cal.get(Calendar.MONTH) + 1,
+                                cal.get(Calendar.DAY_OF_MONTH),
+                                cal.get(Calendar.HOUR_OF_DAY),
+                                cal.get(Calendar.MINUTE),
+                                cal.get(Calendar.SECOND));
+        }
+
+        /// <summary>
+        /// Converts a Java Date value to a UTC DateTime value
+        /// </summary>
+        /// <param name="utcDateTime">The UTC DateTime value</param>
+        /// <returns>A UTC Java Date value</returns>
+        private static Date ToJavaDateUtc(DateTime utcDateTime)
+        {
+            var cal = Calendar.getInstance();
+            cal.setTimeZone(TimeZone.getTimeZone("UTC"));
+
+            cal.set(Calendar.YEAR, utcDateTime.Year);
+            cal.set(Calendar.MONTH, utcDateTime.Month - 1);
+            cal.set(Calendar.DAY_OF_MONTH, utcDateTime.Day);
+            cal.set(Calendar.HOUR_OF_DAY, utcDateTime.Hour);
+            cal.set(Calendar.MINUTE, utcDateTime.Minute);
+            cal.set(Calendar.SECOND, utcDateTime.Second);
+
+            return cal.getTime();
+        }
+
+        /// <summary>
+        /// Checks if downloader can get the data for the symbol
+        /// </summary>
+        /// <param name="symbol"></param>
+        /// <returns>Returns true if the symbol is available</returns>
+        public bool HasSymbol(Symbol symbol)
+        {
+            return _instruments.ContainsKey(symbol);
+        }
+
+        /// <summary>
+        /// Gets the security type for the specified symbol
+        /// </summary>
+        /// <param name="symbol">The symbol</param>
+        /// <returns>The security type</returns>
+        public SecurityType GetSecurityType(Symbol symbol)
+        {
+            return _instruments[symbol].Type;
+        }
+
+        /// <summary>
+        /// Get historical data enumerable for a single symbol, type and resolution given this start and end time (in UTC).
+        /// </summary>
+        /// <param name="symbol">Symbol for the data we're looking for.</param>
+        /// <param name="type">Security type</param>
+        /// <param name="resolution">Resolution of the data request</param>
+        /// <param name="startUtc">Start time of the data in UTC</param>
+        /// <param name="endUtc">End time of the data in UTC</param>
+        /// <returns>Enumerable of base data for this symbol</returns>
+        public IEnumerable<BaseData> Get(Symbol symbol, SecurityType type, Resolution resolution, DateTime startUtc, DateTime endUtc)
+        {
+            if (!_instruments.ContainsKey(symbol))
+                throw new ArgumentException("Invalid symbol requested: " + symbol);
+
+            if (resolution == Resolution.Tick)
+                throw new NotSupportedException("Resolution not available: " + resolution);
```

Comment:
FXCM supports tick, but Oanda doesn't. This might be copy-pasta from Oanda's implementation